pkg/api/store/cluster/cluster_store.go

```diff
@@ -254,6 +352,29 @@ func (r *Store) Update(apiContext *types.APIContext, schema *types.Schema, data
 		}
 	}
 
+	//check if template is passed. if yes, load template data
+	if hasTemplate(data) {
+		if existingCluster[managementv3.ClusterSpecFieldClusterTemplateID] == "" {
+			return nil, fmt.Errorf("this cluster is not created using a template, cannot update it to use a template now")
+		}
+
+		clusterTemplateRevision, err := r.validateTemplateInput(data, true)
+		if err != nil {
+			return nil, err
+		}
+		clusterUpdate, err := loadDataFromTemplate(clusterTemplateRevision, data)
+		if err != nil {
+			return nil, err
+		}
+		updatedTemplateID := convert.ToString(clusterUpdate[managementv3.ClusterSpecFieldClusterTemplateID])
+		templateID := convert.ToString(existingCluster[managementv3.ClusterSpecFieldClusterTemplateID])
+		if !strings.EqualFold(updatedTemplateID, templateID) {
+			return nil, fmt.Errorf("cannot Update Cluster, Cluster cannot be changed to a new template")
```

Comment:
Nit: we should remove caps from the words in the middle of the sentence. Can you also check if error messages in Rancher API errors should start with cap